packages/xod-client/src/core/containers/Catcher.jsx

```diff
@@ -0,0 +1,188 @@
+import * as R from 'ramda';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from 'react-redux';
+import storeShape from 'react-redux/src/utils/storeShape';
+import { Icon } from 'react-fa';
+import cls from 'classnames';
+
+import { recoverState } from '../actions';
+import CloseButton from '../components/CloseButton';
+import { getUtmForumUrl } from '../../utils/urls';
+
+class Catcher extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.timer = null;
+    this.stableState = props.store.getState();
+    this.state = {
+      error: null,
+      errorInfo: null,
+      recovering: false,
+    };
+
+    this.unsubscribe = props.store.subscribe(() => {
+      const nextState = props.store.getState();
+      if (this.stableState !== nextState) {
+        clearTimeout(this.timer);
+        // Postpone memoizing of the stable state on the next
+        // tick after getting updated store state,
+        // because the new store state might cause an error
+        this.timer = setTimeout(() => {
+          this.stableState = nextState;
+        }, 0);
+      }
+    });
+
+    // Ref to the App container is needed to call methods
+    // if there are no errors or got some
+    this.appRef = null;
+
+    // Ref to the Textarea with the error log
+    // is needed to select all on focus
+    this.textareaRef = null;
+
+    this.onClose = this.onClose.bind(this);
+    this.selectAllTextareaContent = this.selectAllTextareaContent.bind(this);
+  }
+
+  componentDidMount() {
+    // Open initial project on the startup
+    this.appRef.refs.wrappedInstance.onFirstRun();
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  onClose() {
+    this.setState({ error: null, errorInfo: null });
+  }
+
+  getErrorReport() {
+    const isActionErrored = !R.equals(
+      this.stableState.lastActions,
+      this.props.store.getState().lastActions
+    );
+
+    const lastOneCausedError = [
+      '# AND THE LAST ONE CAUSED AN ERROR:',
+      '# YOU CAN TRY TO DISPATCH THE LAST ONE',
+      '# TO REPROUCE THE ERROR',
+    ].join('\n');
+
+    const stableStateReport = R.compose(
+      stableState => JSON.stringify(stableState, null, 2),
+      R.assocPath(['project', 'apiKey'], 'SECRET'),
+      R.assocPath(['user', 'grant'], 'SECRET'),
+      R.omit(['lastSavedProject', 'projectHistory'])
+    )(this.stableState);
+
+    return [
+      // Error
+      '# ERROR',
+      this.state.error.stack,
+      '# COMPONENT STACK',
+      this.state.errorInfo.componentStack,
+      '\n\n',
+      '# LAST ACTIONS (max 3 in log)',
+      isActionErrored
+        ? lastOneCausedError
+        : '# BUT NONE OF THEM IS NOT THE REASON',
```

Comment:
Maybe 
```suggestion
        : '# BUT NONE OF THEM IS THE REASON',
```
is more correct?

Or I didn't quite get the point of this message.